discord/ext/converters/regex.py

```diff
@@ -0,0 +1,16 @@
+import re
+from typing import Union
+
+from discord.ext import commands
+
+
+class MatchConverter(commands.Converter):
+    def __init__(self, *, pattern: str, exception: str = "No match found.") -> None:
+        self.pattern = pattern
+        self.exception = exception
+
+    async def convert(self, ctx, argument: str) -> str:
+        match: Union[re.Match, None] = re.match(self.pattern, argument)
```

Comment:
I recall Python's `re.match` is borderline useless, probably use `re.fullmatch` instead.